### PR TITLE
Sync dataflow libraries

### PR DESCRIPTION
**Pull request by owen-mc**
Pulls in https://github.com/github/codeql/pull/5128 and probably some other PRs

### CODE REVIEW

**Comment by smowton** [comment]
Needs autoformat

File: ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll [hunk 1]
```diff
@@ -186,7 +186,7 @@ abstract class DefaultTaintSanitizer extends DataFlow::Node { }
  * Holds if `node` should be a sanitizer in all global taint flow configurations
  * but not in local taint.
  */
-predicate isDefaultTaintSanitizer(DataFlow::Node node) { node instanceof DefaultTaintSanitizer }
+predicate defaultTaintSanitizer(DataFlow::Node node) { node instanceof DefaultTaintSanitizer }
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll:189]: guess this is due to a clash? Make the DefaultTaintSanitizerGuard predicate consistent?

Comment (owen-mc) [reply ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll:189]: I updated the name from `defaultTaintSanitizer` to `isDefaultTaintSanitizer` a while ago, but I hadn't appreciated it was used in a file that gets synced, so I've reverted the change to make it easier to sync.

I've now renamed `isDefaultTaintSanitizerGuard` to `defaultTaintSanitizerGuard` to be consistent.

**Comment by owen-mc** [comment]
The dist-compare looked fine except `aws-sdk-go`, which took 5% longer despite being quite a large repo. I re-ran it and this time it took 6% less time, so it seems highly unlikely that there's a real performance change.

**Review by sauyon — APPROVED** [review]
Ah, this was indeed quite trivial.

**Pull request merged**